src/main.rs

```diff
@@ -72,26 +66,23 @@ async fn main() {
         namespace: matches.value_of("cluster_namespace").unwrap_or_default(),
     };
 
-    let deploy_method = matches.value_of("deploy_method").unwrap();
-    let local_path = matches.value_of("local-path");
-    match deploy_method {
-        method if method == DeployMethod::Local.to_string() => {
-            if local_path.is_none() {
-                panic!("Error: --local-path is required for 'local' deploy-method.");
-            }
-        }
-        _ => {
-            if local_path.is_some() {
-                warn!("WARN: --local-path <path> will be ignored");
-            }
-        }
-    }
+    let deploy_method = if let Some(local_path) = matches.value_of("local_path") {
+        DeployMethod::Local(local_path.to_owned())
+    } else if let Some(release_channel) = matches.value_of("release_channel") {
+        DeployMethod::ReleaseChannel(release_channel.to_owned())
+    } else {
+        unreachable!("One of --local-path or --release-channel must be provided.");
+    };
 
-    let solana_root = match matches.value_of("local-path") {
-        Some(path) => SolanaRoot::new_from_path(path.into()),
-        None => SolanaRoot::default(),
+    let solana_root = match &deploy_method {
+        DeployMethod::Local(path) => SolanaRoot::new_from_path(path.into()),
+        DeployMethod::ReleaseChannel(_) => SolanaRoot::default(),
     };
 
+    let build_type: BuildType = matches
+        .value_of_t("build_type")
+        .unwrap_or_else(|e| e.exit());
```

Comment:
micro-nit: you should be able to just `unwrap()` here since it has a default value